staging/src/k8s.io/apiserver/pkg/endpoints/handlers/update.go

```diff
@@ -101,6 +102,30 @@ func UpdateResource(r rest.Updater, scope RequestScope, admit admission.Interfac
 				return newObj, mutatingAdmission.Admit(admission.NewAttributesRecord(newObj, oldObj, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo))
 			})
 		}
+		transformers = append(transformers, func(ctx context.Context, newObj, oldObj runtime.Object) (runtime.Object, error) {
+			if hasUID, err := hasUID(oldObj); err != nil {
+				return nil, err
+			} else if !hasUID && scope.Authorizer != nil {
```

Comment:
skipping this check if no authorizer is provided seems like an opportunity for misconstruction to leave a security hole. can we require scope.Authorizer to be set? as far as I'm aware, every handler (except probably unit tests :-/) has at least an "AlwaysAllow" authorizer implementation set

Reply:
When the apiserver is coming up, it is allowed for the authorizer to be unset, it just emits a warning saying authorization is disabled. I think if we allow that at startup then we should allow for this, maybe we could emit a warning when the handler is registered too, if the authorizer is unset?

Reply:
> When the apiserver is coming up, it is allowed for the authorizer to be unset, it just emits a warning saying authorization is disabled. I think if we allow that at startup then we should allow for this

Hmm, it still seems safer for the top level server creator to plug in an AlwaysAllow impl in that case. Skipping on nil here means we don't know if we were wired incorrectly or if the server creator really intended to run without authorization.